servers/jax-rs-tests/src/main/java/org/projectnessie/jaxrs/gc/AbstractRestGCTest.java

```diff
@@ -0,0 +1,427 @@
+    // two commits for "table_1" on branch1
+    CommitOutput table1 =
+        commitSingleOp(
+            prefix, branch1, branch1.getHash(), 42, "cid_1", "table_1", "file1", null, null);
+    table1 =
+        commitSingleOp(
+            prefix, branch1, table1.hash, 43, "cid_1", "table_1", "file2", table1.content, null);
+    // two commits for "table_2" on branch1
+    CommitOutput table2 =
+        commitSingleOp(
+            prefix, branch1, branch1.getHash(), 42, "cid_2", "table_2", "file1", null, null);
+    table2 =
+        commitSingleOp(
+            prefix, branch1, table2.hash, 43, "cid_2", "table_2", "file2", table2.content, null);
+    // All 4 commits are expected to be expired as it is before the cutoff time and there are some
+    // more commits for
+    // each table after cutoff timestamp (i,e these 4 commits are not head commits)
+    fillExpectedContents(branch1, 4, expectedExpired);
+
+    final Instant cutoffTime = Instant.now();
+
+    // two commit for "table_1" on branch1
+    table1 =
+        commitSingleOp(
+            prefix, branch1, table1.hash, 44, "cid_1", "table_1", "file3", table1.content, null);
+    table1 =
+        commitSingleOp(
+            prefix, branch1, table1.hash, 45, "cid_1", "table_1", "file4", table1.content, null);
+
+    // one commit for "table_2" on branch1
+    table2 =
+        commitSingleOp(
+            prefix, branch1, table2.hash, 44, "cid_2", "table_2", "file3", table2.content, null);
+    // drop table "table_2"
+    dropTableCommit(prefix, branch1, table2.hash, "table_2");
+    // table_2's one commit (after cutoff timestamp) is also expected to be expired as table is
+    // dropped above.
+    fillExpectedContents(branch1, 2, expectedExpired);
+
+    performGc(
+        cutoffTime,
+        null,
+        expectedExpired,
+        new HashSet<>(Arrays.asList(prefix + "_cid_1", prefix + "_cid_2")),
+        true);
+  }
+
+  @Test
+  public void testSingleRefRenameTable() throws BaseNessieClientServerException {
+    String prefix = "singleRefRenameTable";
+    Map<String, Set<Content>> expectedExpired = new HashMap<>();
+
+    Branch branch1 = createBranch(prefix);
+    // one commit for "table_2" on branch1
+    CommitOutput table2 =
+        commitSingleOp(
+            prefix, branch1, branch1.getHash(), 42, "cid_2", "table_2", "file1", null, null);
+    // expect this commit to be expired as it is before cutoff time.
+    fillExpectedContents(branch1, 1, expectedExpired);
+
+    final Instant cutoffTime = Instant.now();
+
+    // two commit for "table_2" on branch1
+    table2 =
+        commitSingleOp(
+            prefix, branch1, table2.hash, 43, "cid_2", "table_2", "file2", table2.content, null);
+    table2 =
+        commitSingleOp(
+            prefix, branch1, table2.hash, 44, "cid_2", "table_2", "file3", table2.content, null);
+
+    // rename table "table_2" to "table_2_renamed" on branch1
+    table2 =
+        commitSingleOp(
```

Comment:
This will result in `cid_2` being accessible as `table_2` *and* `table_2_renamed`.
Aka: the commit for the table-rename here must include a `Delete` operation for `table_2` and a `Put` operation for `table_2_renamed`

Reply:
It is handled already. I am passing `beforeRename` variable and based on that it is `put` and `delete` inside. other cases it is just `put`.